prov/gni/src/gnix_vc.c

```diff
@@ -2049,12 +2054,17 @@ int _gnix_vc_nic_progress(struct gnix_nic *nic)
 			GNIX_DEBUG(FI_LOG_EP_CTRL,
 				   "__gnix_vc_push_work_reqs failed: %d\n", ret);
 
-		ret = __gnix_vc_push_tx_reqs(vc);
-		if (ret != FI_SUCCESS)
+
+		ret_tx = __gnix_vc_push_tx_reqs(vc);
+		if (ret_tx != FI_SUCCESS) {
```

Comment:
Can you move this debug statement into the if (ret_tx != FI_SUCCESS) below the release of the lock? It would clean it up slightly.